spec/cookbook_spec.rb

```diff
@@ -170,18 +180,64 @@
       :result => [
       ],
     },
+    {
+      :name => 'when metadata file is not in the root of the cb dir',
+      :files => [
+        {
+          :status => :deleted,
+          :path => 'cookbooks/two/cb_one/files/default/metadata.rb',
+        },
+      ],
+      :result => [
+        ['cb_one', :modified],
+      ],
+    },
   ]
 
-  fixtures.each do |fixture|
-    it "should handle #{fixture[:name]}" do
-      BetweenMeals::Changes::Cookbook.find(
-        fixture[:files],
-        cookbook_dirs,
-        logger,
-      ).map do |cb|
-        [cb.name, cb.status]
-      end.
-        should eq(fixture[:result])
+  {
+    'Normally' => false,
+    'With symlinks' => true,
+  }.each do |c, track_symlinks|
+    context "Running BetweenMeals #{c}" do
+      before do
+        allow(File).to receive(:realpath).with(repo_path).and_return(repo_path)
```

Comment:
You're testing that with-symlink works with symlinks and not with-symlinks works without symlinks, but you're not testing if not-with-symlinks works even when there are symlinks.

Reply:
that doesn't work today. Since symlinks will never have a trailing '/' they won't pass the regex in explode_path ```re = %r{^#{dir}/([^/]+)/.*}```

Reply:
I know it doesn't work today... but that's the point, if you're going to add support for turning on this new feature, make sure there's a test to not allow it to be on if the user didn't request it.

Reply:
addressed